server/src/main/java/org/elasticsearch/index/mapper/DocumentMapper.java

```diff
@@ -120,6 +122,8 @@ public DocumentMapper build(MapperService mapperService) {
     private final MetadataFieldMapper[] deleteTombstoneMetadataFieldMappers;
     private final MetadataFieldMapper[] noopTombstoneMetadataFieldMappers;
 
+    private static final Pattern NAME_WITH_WHITESPACE_CHARACTERS = Pattern.compile(".*(?:[\\t\\r\\n]+.*)+");
```

Comment:
Is there any way to do this without a none-capturing group ?

Reply:
Maybe if you replace all whitespaces with `" "` and check if the strings are the same, but probably isn't going to be faster: `replaceAll("\\p{Zs}+", " ");`

But isn't this problematic if someone uses other weird chars like `\a` or other weird unicode chars?

Reply:
The bell operator `\a` and other weird Unicode chars will not pass the Jackson parser validation. Whitespace characters are valid as JSON keys, but result in weird schemas when used in inner columns.

Reply:
Could iterate over the characters of the string and check them?

Reply:
> Could iterate over the characters of the string and check them?

Could be the faster way, maybe try a jmh benchmark  to check those alternatives.

Reply:
Good idea, i will try that.

Reply:
```
Benchmark                                         Mode  Cnt        Score        Error  Units
ColumnNameBenchmark.measure_validate_loop_chars  thrpt   25  7945104.819 ± 518013.577  ops/s
ColumnNameBenchmark.measure_validate_regex       thrpt   25   250635.866 ±   3263.536  ops/s
```
https://gist.github.com/mkleen/bdeff74d3be62890cd5e40a571bb147a
Numbers are clear hear, let's not use a regex ;-)